tests/components/decora_wifi/test_common.py

```diff
@@ -0,0 +1,223 @@
+"""Test the decora_wifi Common components."""
+
+import logging
+from unittest.mock import patch
+
+from homeassistant.components.decora_wifi.common import (
+    CommFailed,
+    DecoraWifiEntity,
+    DecoraWifiPlatform,
+    LoginFailed,
+)
+from homeassistant.components.decora_wifi.const import DOMAIN
+from homeassistant.helpers import device_registry
+
+from tests.components.decora_wifi.common import (
+    SWITCH_NAME,
+    FakeDecoraWiFiIotSwitch,
+    FakeDecoraWiFiResidence,
+    FakeDecoraWiFiResidentialAccount,
+    FakeDecoraWiFiSession,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+USERNAME = "username@home-assisant.com"
+PASSWORD = "test-password"
+INCORRECT_PASSWORD = "incoreect-password"
+MODEL = "DW4SF"
+
+
+async def test_DecoraWifiPlatform_init(hass):
```

Comment:
https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Don't patch core code. Patch the `pypi` package code instead.